packages/backend/src/open_payments/payment_pointer/service.test.ts

```diff
@@ -133,6 +134,107 @@ describe('Open Payments Payment Pointer Service', (): void => {
     })
   })
 
+  describe('Update Payment Pointer', (): void => {
+    test.each`
+      initialIsActive | status                           | publicName
+      ${true}         | ${undefined}                     | ${'Public Name 1'}
+      ${true}         | ${PaymentPointerStatus.Inactive} | ${'Public Name 2'}
+      ${false}        | ${PaymentPointerStatus.Active}   | ${null}
+      ${false}        | ${undefined}                     | ${null}
+    `(
+      'Payment pointer  with initial isActive of $initialIsActive can be updated with status of $status and publicName of $publicName',
+      async ({ initialIsActive, status, publicName }): Promise<void> => {
+        const paymentPointer = await createPaymentPointer(deps)
+
+        if (!initialIsActive) {
+          await paymentPointer.$query(knex).patch({ deactivatedAt: new Date() })
+        }
+
+        const updatedPaymentPointer = await paymentPointerService.update({
+          id: paymentPointer.id,
+          status,
+          publicName
+        })
+        assert.ok(!isPaymentPointerError(updatedPaymentPointer))
+
+        expect(updatedPaymentPointer.publicName).toEqual(publicName)
+
+        // check status. should be whatever the update input is, or the original value if not updated
+        let statusChecked = false
```

Comment:
I feel like we can simplify this test by:
- having an `expectedStatus` column in the table
- removing `publicName` since IMO it's tested sufficiently in "Payment pointer fields can be updated independently > publicName"

Reply:
There isn't really a `status` to expect at this level but there is an `isActive`. I think having `expectedIsActive` is good all the same instead of determining the output based off the combination of `status`/`initialIsActive`. And I take your point with `publicName` - it's not really relevant to what we want to test here.

As for not having a status to expect here, we are getting a status from the gql input and passing that into the update. What we get from the database just has an isActive (via the model). Maybe we should pass in `isActive` instead? For the current use case that seemed unnecessary because we're just passing in the gql input and not getting it from some payment pointer record with `paymentPointer.isActive` or something. But I guess we might want to. Additionally, the way it is now also makes the test a bit more confusing to read (and maybe makes the update method more confusing to think about).